### DISCUSSION TITLE
Next 10 Internationalized Routing not using other locale

### DISCUSSION THREAD

**Discussion by idrisakmal** [vercel/next.js#18615] [Help]
# Help Wanted

## Describe the bug

I have upgraded my app to next@10.0.0 to use the Internationalized Routing feature. I have zh and en as the available locales and running `react-i18next` as the localization tool. My default locale is set to zh and fallback to en which means when I navigate to `/` it will use zh and `/en` will supposedly use english. 

`/` works fine as it should but when going to the subpath `/en` it doesn't use my english translation file and still shows the `zh` language. 

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

https://codesandbox.io/s/gifted-cohen-7v86n?file=/pages/index.js

## Expected behavior

I expect users to browse to www.example.com/en to get the english version of the site.

## System information

- OS: WSL2 Ubuntu 20.04
- Browser chrome, safari
- Version of Next.js: 10.0.0
- Version of Node.js: 10.18.1

**Comment by idrisakmal** [comment]
Codes:

i18n.js

```import i18n from "i18next";
import { initReactI18next } from "react-i18next";

import en from "./public/locales/en/common.json";
import zh from "./public/locales/zh/common.json";

i18n
  .use(initReactI18next) // passes i18n down to react-i18next
  .init({
    resources: {
      en: {
        translation: en,
      },
      zh: {
        translation: zh,
      },
    },
    lng: "zh",

    interpolation: {
      escapeValue: false,
    },
  });

export default i18n;
```
next.config.js
```
module.exports = {
  i18n: {
    locales: ["en", "zh"],
    defaultLocale: "zh",
  },
};
```

**Comment by timneutkens** [comment]
Converted to a help discussion given this does not sound like a bug in Next.js as we only handle routing and you said routing is working correctly. You'll probably have to use https://react.i18next.com/legacy-v9/i18nextprovider and pass `initialLanguage`.

It would look something like this (did not test it):

```js
// pages/_app.js
import {useRouter} from 'next/router'
import { I18nextProvider } from 'react-i18next';

function MyApp({ Component, pageProps }) {
  const router = useRouter()
  return <I18nextProvider i18n={ i18n } initialLanguage={router.locale}>
    <Component {...pageProps} />
  </I18nextProvider>
}

export default MyApp
```

**Reply by idrisakmal** [reply]
sorry @timneutkens yes suppose I need help. i'll try that out. thanks!

**Reply by vinzloh** [reply]
https://codesandbox.io/s/purple-cherry-6o7o0?file=/pages/index.js

just need to hook up with 18next 😉

**Reply by xairoo** [reply]
Router should pass the language directly into i18n.init().
Otherwise the default language will be loaded and shown before it gets changed if the lang is set at the URL.

For example calling (a refresh shows it fine) http://localhost:3000/en
will result in a visible header change from `Great Title in Chinese Simplified` to `Great Title in English` after the site is loaded.

**Comment by idrisakmal** [comment]
ayyy dude @vinzloh that works out beautifully. need to notify react-i18next for the language change. thanks for helping mate!

thank you NextJS 10 too for this new routing feature. its awesome!

**Comment by Daidalos117** [comment]
Shame that this is not mentioned in docs. A also had trouble to understand how routes and i18 connect to each other.
